src/DetermineSampleTimestamp.hpp

```diff
@@ -0,0 +1,45 @@
+#ifndef _AGGREGATOR_DETERMINE_SAMPLE_TIMESTAMP_HPP_
+#define _AGGREGATOR_DETERMINE_SAMPLE_TIMESTAMP_HPP_
+
+#include <base/Time.hpp>
+#include <boost/shared_ptr.hpp>
+#include <stdexcept>
+
+namespace base { namespace samples
```

Comment:
Since this is really something specific to the aggregator ... I would put it in the aggregator namespace.